plugins/CoreHome/ChangesHelper.php

```diff
@@ -0,0 +1,142 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Plugins\CoreHome;
+
+use Piwik\Common;
+use Piwik\Date;
+use Piwik\Db;
+use Piwik\Piwik;
+use Piwik\Plugin\Manager;
+use Piwik\Plugins\UsersManager\Model AS UsersModel;
+
+/**
+ * CoreHome changes helper class
+ */
+class ChangesHelper
```

Comment:
Thinking more about that it might even be better to separate things a bit more. So there is one model/dao class that handles the changes entries, having methods like `addChanges`, `getChanges`, `getChangesAfter`, ... and that class is than used by some state class that provides methods like `getChangesForUser` ,`loadPluginChanges`,... If both classes are using dependency injection it will also be possible to write tests for those methods easily, by using mock objects.

Reply:
I've reworked the class structure to use DI rather than static methods. There are now two classes `\Piwik\Changes\Model` which handles all the change data operations independent from any user context and `\Piwik\Changes\UserChanges` which retrieves changes and updates last viewed settings in the context of a supplied user model.